From 72966269d9377d7bf8792d41de0edd6d756fea68 Mon Sep 17 00:00:00 2001
From: zachary zhang <zhangzg@marvell.com>
Date: Tue, 15 May 2018 17:42:43 +0800
Subject: [PATCH 0736/1200] hw_info: mvebu: add support to load HW info when
 env reset

- Call cmd_hw_info_load during env reset to initialize the
  HW configuration.

Change-Id: Iff7e1c42d638f7e0dd34c905ccad3f7c1abbf857
Signed-off-by: zachary zhang <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/55277
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: Igal Liberman <igall@marvell.com>
---
 env/common.c          | 12 ++++++++++++
 include/environment.h |  5 +++++
 2 files changed, 17 insertions(+)

diff --git a/env/common.c b/env/common.c
index f21ff70096..6da5dc5e2c 100644
--- a/env/common.c
+++ b/env/common.c
@@ -88,6 +88,18 @@ void set_default_env(const char *s)
 
 	gd->flags |= GD_FLG_ENV_READY;
 	gd->flags |= GD_FLG_ENV_DEFAULT;
+
+#ifdef CONFIG_CMD_MVEBU_HW_INFO
+		/* load the HW configuration from EEPROM to env variables and
+		 * saveenv.
+		 * This is because when the env varibles are reset, need to
+		 * recover the HW configuration related env varibles from
+		 * EEPROM.
+		 * The only generic U-Boot env variable that will be overwrote
+		 * here will be the Marvell specific variables
+		 */
+		cmd_hw_info_load(NULL, 1);
+#endif /* CONFIG_CMD_MVEBU_HW_INFO */
 }
 
 
diff --git a/include/environment.h b/include/environment.h
index 7986a24277..b8448f80eb 100644
--- a/include/environment.h
+++ b/include/environment.h
@@ -272,6 +272,11 @@ void env_crc_update(void);
 /* Look up the variable from the default environment */
 char *env_get_default(const char *name);
 
+/* load box information to environment */
+#ifdef CONFIG_CMD_MVEBU_HW_INFO
+int cmd_hw_info_load(char *name, int silence);
+#endif /* CONFIG_CMD_MVEBU_HW_INFO */
+
 /* [re]set to the default environment */
 void set_default_env(const char *s);
 
-- 
2.22.0

